addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.xml

```diff
@@ -3,187 +3,56 @@
-                            </span>
-                        </a>
-                        <textarea t-if="(columnIsProductAndLabel.value and label) or (!columnIsProductAndLabel.value and !productName and label)"
-                            class="o_input d-print-none border-0 fst-italic"
-                            placeholder="Enter a description"
-                            readonly="1"
-                            rows="1"
-                            type="text"
-                            t-att-class="sectionAndNoteClasses"
-                            t-att-value="label"
-                            t-on-change="(ev) => this.updateLabel(ev.target.value)"
-                            t-ref="labelNodeRef"
-                        />
-                    </t>
-                </t>
-                <t t-else="">
-                    <t t-if="props.isSection">
-                        <input
-                            class="o_input text-wrap border-0 fst-italic"
-                            placeholder="Enter a description"
-                            type="text"
-                            t-att-class="sectionAndNoteClasses"
-                            t-att-value="label"
-                            t-on-change="(ev) => this.updateLabel(ev.target.value)"
-                            t-ref="labelNodeRef"
-                        />
-                    </t>
-                    <t t-else="">
-                        <textarea
-                            class="o_input d-print-none border-0 fst-italic"
-                            placeholder="Enter a description"
-                            rows="1"
-                            type="text"
-                            t-att-class="sectionAndNoteClasses"
-                            t-att-value="label"
-                            t-on-change="(ev) => this.updateLabel(ev.target.value)"
-                            t-ref="labelNodeRef"
-                        />
-                    </t>
-                </t>
+            <t t-if="isNote">
+                <textarea
+                    class="o_input d-print-none border-0 fst-italic"
+                    placeholder="Enter a description"
+                    rows="1"
+                    t-att-class="sectionAndNoteClasses"
+                    t-att-value="label"
+                    t-on-change="(ev) => this.updateLabel(ev.target.value)"
+                    t-ref="labelNodeRef"
+                />
+            </t>
+            <t t-elif="isSection">
+                <input
+                    type="text"
+                    class="o_input text-wrap border-0 fst-italic fw-bold"
+                    placeholder="Enter a description"
+                    t-att-class="sectionAndNoteClasses"
+                    t-att-value="label"
+                    t-on-change="(ev) => this.updateLabel(ev.target.value)"
+                    t-ref="labelNodeRef"
+                />
```

Comment:
Why do you stop considering the readonly props here ?
In sale, there is a setting to lock confirmed orders, which shouldn't be modifiable anymore, but in your branch, the description on the lines (including notes and sections) can still be edited when the order is locked (and the `order_line` field is readonly)

Reply:
Section and note were also input and textarea even when the line was readonly.
The PR does not change these cases but yep, I didn't manage the case of the readonly product description for confirmed orders. This should be fixed now.

I tried to factorize this template because it was really complicated to understand what is displayed and when.
Section and note was the same in readonly and edit, the difference was "just" a placeholder in edit.
When it was not a section nor a note, we had a span or the Many2XAutocomplete with the textarea for the product description. I replaced this big part by using the Many2One which display a span in readonly and the Many2XAutocomplete in edit + the textarea.